app/javascript/article-form/components/ArticleCoverImage.jsx

```diff
@@ -17,14 +19,16 @@ export class ArticleCoverImage extends Component {
     this.setState({ uploadingImage: false });
   };
 
-  handleMainImageUpload = (e) => {
-    e.preventDefault();
+  handleMainImageUpload = (event) => {
+    event.preventDefault();
 
     this.setState({ uploadingImage: true });
     this.clearUploadError();
 
     if (validateFileInputs()) {
-      const payload = { image: e.target.files, wrap_cloudinary: true };
+      const { files: image } =
+        event instanceof DragEvent ? event.dataTransfer : event.target;
```

Comment:
Need to see what type of event it is so we can grab the appropriate `files` property.